src/Random/Distribution.h

```diff
@@ -0,0 +1,126 @@
+// -*- C++ -*-
+/***************************************************************************
+ *
+ * The IPPL Framework
+ *
+ * This program was prepared by PSI.
+ * All rights in the program are reserved by PSI.
+ * Neither PSI nor the author(s)
+ * makes any warranty, express or implied, or assumes any liability or
+ * responsibility for the use of this software
+ *
+ * Visit www.amas.web.psi for more details
+ *
+ ***************************************************************************/
+
+// Class Distribution
+//   This class can be used for creating a distribution object
+//   with custom pdf, cdf, and estimate function that is used for
+//   the sampling method.
+//
+
+#ifndef IPPL_DISTRIBUTION_H
+#define IPPL_DISTRIBUTION_H
+
+#include "Random/Utility.h"
+#include "Types/ViewTypes.h"
+
+namespace ippl {
+  namespace random {
+  
+    /*!
+     * @file Distribution.h
+     * @class Distribution
+    */
+  
+   /*!
+    * @ingroup Distribution
+    * @brief A class that represents a distribution.
+    * @tparam T Datatype.
+    * @tparam Dim Dimensionality of sample space.
+    * @tparam DimP Dimensionality of the parameter array.
+    * @tparam PDF Struct type for the PDF (Probability Density Function).
+    * @tparam CDF Struct type for the CDF (Cumulative Distribution Function).
+    * @tparam ESTIMATE Struct type for the ESTIMATE function.
+    */
+    template <typename T, unsigned Dim, unsigned DimP, typename PDF, typename CDF, typename ESTIMATE>
```

Comment:
We need T for datatype of parameters.
We need Dim because the input argument of getFullPdf is a ippl vector that takes T and Dim as template arguments.
We need DimP for allocating an array of parameters.
We need to pass PDF, CDF and ESTIMATE as functors since function pointers don't work on gpus. I guess we can combine them in a single functor if that's preferable.

Reply:
I combined pdf/cdf/estimate into one struct. Please see b0ae163 and let me know what you think.